src/fabric.c

```diff
@@ -344,6 +344,8 @@ void fi_ini(void)
 {
 	char *param_val = NULL;
 
+	ofi_osd_init();
```

Comment:
Yes, it is possible, but in this case ini_lock should be global variable, not static. If you ok to change visibility scope then I can easily make such fix